test.py

```diff
@@ -78,6 +78,15 @@ def nocolor(text: str) -> str:
         return palette.ansi_escape.sub('', text)
 
 
+class TestCase(NamedTuple):
+    suite_name: str
+    name: str
+    case: Optional[str]
+
+    def __str__(self):
+        return f"{self.suite_name}::{self.name}{'::' + self.case if self.case else ''}"
```

Comment:
Note: Pytest usually uses "filename.py" before the "::". I think (but I'm not sure) your self.name drops the ".py". It's not terrible, but we should be aware of this difference.
Also, despite my recommendation, it seems some suites used classes inside the test files, so the test path gains yet another component - the class name.  I guess this isn't supported here? (I'm not too sad, I think these test classes are a mistake anyway).

Reply:
Makes sense.

Reply:
By the way, some of the source files have a path, e.g., cql-pytest's `cassandra_tests/validation/entities/type_test.py`. Is this full path in self.name (it should...)?

Reply:
I don't see what pytest has to do with boost or unit tests, so we can't support what pytest supports anyway. cql tests don't have test cases, at least not now.

Reply:
In my [case PR](https://github.com/scylladb/scylladb/pull/15238)  when there's no case it's omitted like `suite::test_file`.

Reply:
@kostja I don't understand your question. Let me try to explain, hoping it's related to your question:
test.py can run different types of tests - pytest tests, boost tests, cql approval tests, and so on. So we're looking for some uniform syntax for representing the name of a test to run (whole suite, whole file, or just function, whenever that's possible) that can be inspired by the syntax used by one of the above frameworks (I propose pytest) but needs to be translated to each.

So for example `cql-pytest::test_uda.py::test_custom_avg` can refer to one test function in a cql-pytest, `boost::wasm_test::test_long_udf_yields` can refer to one test function in Boost, and so on.

My comment above was that the test-file part of a cql-pytest can include the ".py" (I guess we can make it optional) and also be a pathname with slashes in it, like `cassandra_tests/validation/entities/type_test.py`. Elsewhere I also noted that in dtest (but I think in none of the scylla.git frameworks?) pytest tests can also be inside classes, you you have yet another component between the file name and test function name - the class name.

Reply:
Fixed in https://github.com/scylladb/scylladb/commit/38de3c067b5f6e2bce99011c55710ca92ffc4968